### PR TITLE
Require Java 11 or later for all gradle tasks

### PR DESCRIPTION
**Pull request by hsinn0**
[#184142402]

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -73,3 +73,5 @@ subprojects {
 
 // Override spring boot's kotllin version dependency
 ext['kotlin.version'] = '${kotlinVersion}'
```
Comment (swalchemist) [inline build.gradle:75]: Does the Kotlin version also need to be upgraded?

File: build.gradle [hunk 2]
```diff
@@ -73,3 +73,5 @@ subprojects {
...
+
+assert JavaVersion.current().isJava11Compatible()
```
Comment (swalchemist) [inline build.gradle:77]: Hmmm, surprising that we didn't already having something like this for Java 8.

**Review by swalchemist — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (hsinn0) [reply build.gradle:75]: > Does the Kotlin version also need to be upgraded?

No, and we already tested the build with Java 11 and current Kotlin version.

File: build.gradle (hunk 2 above)
Comment (hsinn0) [reply build.gradle:77]: The build probably had failed with JDK 1.7 anyhow, so no need to check, I guess. Right now, we can build the project with Java 8 just fine AFAICT. This assertion will block using JDK/JRE 8 for any gradle tasks, so prevents running into any possible untested issues later at runtime, etc.

**Pull request merged**